packages/eslint-plugin/docs/rules/no-require-imports.mdx

```diff
@@ -90,10 +92,22 @@ import foo from 'foo';
 </TabItem>
 </Tabs>
 
+## Usage with CommonJS
+
+While this rule is primarily intended to promote ES Module syntax, it still makes sense to enable this rule when authoring CommonJS modules.
+
+If you prefer to use TypeScript's ES Module syntax, which is transformed to `require()` calls during transpilation when outputting CommonJS, you can simply use the rule's default behavior.
```

Comment:
[Phrasing] _"ES Module syntax"_ isn't known to everyone, and _"simply"_ accidentally implies easiness.

```suggestion
If you prefer to use TypeScript's built-in `import ... from ...` ES Module syntax, which is transformed to `require()` calls during transpilation when outputting CommonJS, you can use the rule's default behavior.
```